tests/unit/cluster/failover.tcl

```diff
@@ -70,3 +70,64 @@ test "Instance #0 gets converted into a slave" {
 }
 
 } ;# start_cluster
+
+# Add all replicas to the first primary node.
+proc cluster_allocate_replicas_to_one_primary {primaries replicas} {
+    set primary_id 0
+    for {set j 0} {$j < $replicas} {incr j} {
+        set replica_id [cluster_find_available_replica $primaries]
+        set primary_myself [cluster_get_myself $primary_id]
+        R $replica_id cluster replicate [dict get $primary_myself id]
+    }
+}
+
+start_cluster 5 10 {tags {external:skip cluster} overrides {cluster-node-timeout 5000}} {
```

Comment:
Can't we just have a single shard cluster for this test instead of using the above proc `cluster_allocate_replicas_to_one_primary`?

Reply:
a single shard cluster does not have the enough vote to do the failover

Reply:
Yeah, my bad. Do we need to transfer the replica(s) to a single primary?

Reply:
sorry, what do you means? i did not get the point (bad english)

Reply:
I'm not sure why do we need to move all the replicas to a single primary. If it's required, could you add a comment stating the reason for it ?

Reply:
Strictly speaking, it is not necessary. I want to save the total number of nodes, and the replicas are move to a single primary just for rank.

Reply:
Yeah, I feel it's unnecessary. If required, you could initialize with more replicas in each shard.

Reply:
ok, i get the point, yes, it is unnecessary, i was thinking of having a test case to verify the change at that time, now i updated the test code